datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxRetryService.java

```diff
@@ -38,7 +41,13 @@
     private StackV4Endpoint stackV4Endpoint;
 
     public FlowIdentifier retrySdx(SdxCluster sdxCluster) {
-        return flow2Handler.retryLastFailedFlow(sdxCluster.getId(), lastSuccessfulStateLog -> retryCloudbreakIfNecessary(sdxCluster, lastSuccessfulStateLog));
+        FlowLog flowLog = flow2Handler.getFirstStateLogfromLatestFlow(sdxCluster.getId());
+        if (getFullRetryFlows().contains(flowLog.getFlowType().getClassValue())) {
```

Comment:
It is possible that a flow type is removed. Then we will get a class not found exception. Please add an outer if condition or a totally separate if condition that throws internal server error if the `flowLog.getFlowType()` is not on classpath

```java
if(!flowLog.getFlowType().isOnClassPath()) {
  throw new InternalServerErrorException(<Flow is not supported please contact support message...>)
}
```

Reply:
will fix.